### PR TITLE
Add locale converter to enhanced DynamoDB client

### PR DESCRIPTION
**Pull request by thomasturrell**
## Motivation and Context
The Java Locale is currently not supported by the DynamoDB client.

## Modifications
Used UriAttributeConverter as a template to create LocaleAttributeConverter.

## Testing
Added tests and tested locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by thomasturrell** [comment]
@debora-ito could you help me with the process to get this pull request reviewed and hopefully merged? I have not contributed to this project before.

**Comment by debora-ito** [comment]
Yes! Sorry for the delay, will bring this up to the team.

**Comment by debora-ito** [comment]
@thomasturrell let us know if you want to continue with this review.

**Comment by thomasturrell** [comment]
> @thomasturrell let us know if you want to continue with this review.

I will fix the issue and return to you. Thank you for the reminder.

File: services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/converters/attribute/StringAttributeConvertersTest.java [hunk 1]
```diff
@@ -178,6 +180,15 @@ public void stringBufferAttributeConverterBehaves() {
         assertThat(transformTo(converter, fromNumber("42")).toString()).isEqualTo("42");
     }
 
+    @Test
+    public void localeAttributeConverterBehaves() {
+        LocaleAttributeConverter converter = LocaleAttributeConverter.create();
+
+        assertThat(transformFrom(converter, Locale.US).s()).isEqualTo(Locale.US);
+
+        assertThat(transformTo(converter, fromString("en-US"))).isEqualTo(Locale.US);
```
[the remark this answers is not in the document]
Comment (thomasturrell) [reply services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/converters/attribute/StringAttributeConvertersTest.java:189]: @zoewangg thank you for taking the time to review my pull request. I am sorry that I did not see this failing test before I opened the pull request.

I have taken sometime to look at this issue. I believe that the test was wrong and the implementation is correct. I have altered the test.

**Comment by thomasturrell** [comment]
@debora-ito I believe that the failing test is now fixed.

**Comment by zoewangg** [comment]
Hi @thomasturrell thank you for addressing the comment. We will take a look shortly.

**Comment by thomasturrell** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=aws_aws-sdk-java-v2&pullRequest=3333)
> 
> [Bug](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&pullRequest=3333&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=aws_aws-sdk-java-v2&pullRequest=3333&metric=new_coverage&view=list) [96.2% Coverage](https://sonarcloud.io/component_measures?id=aws_aws-sdk-java-v2&pullRequest=3333&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=aws_aws-sdk-java-v2&pullRequest=3333&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=aws_aws-sdk-java-v2&pullRequest=3333&metric=new_duplicated_lines_density&view=list)

I believe that the code smell is acceptable because the public modifier is used in all of the other tests in this class. In my opinion, this new test should be consist with the other tests. I suggest that the unnecessary public modifiers should be removed in a separate pull request.

**Review by zoewangg — APPROVED** [review]
Thank you for your contribution!

**Comment by debora-ito** [comment]
@all-contributors please add @thomasturrell for code.

**Pull request merged**